zoe/covid-tracker-react-native
# Update App modal and logic

# Description

Set as Draft until we can prettify the modal (and address any UX concerns).

Mechanism to show a "please update" modal on the app when we deem the user should.
This is based on a Responsible Developer code, in which we only update the API middle version number when we introduce a breaking change. The table below shows this:

```
BE      FE       Show modal?
2.4.0   2.5.0    No
2.4.0   2.4.3    No
2.4.0   2.4.0    No
2.4.0   2.3.99   Yes
2.4.0   1.3.0    Yes
```

Note that the modal won't be shown if the app's version gets ahead of the API.

**How to test:**
Locally, this is very easy:  go into the django settings and set `API_VERSION` to `9.9.9`. Refresh your app ("r" on the console). Modal should appear. 
If using staging, you can hardcode the version to trigger and test the UI: find this method and make it return`true` `checkVersionOfAPIAndApp`

## On what platform have you tested the change?

- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

ugly screenshot. I am in talks with Christina and awaiting a bit of direction from the new designer.
![image](https://user-images.githubusercontent.com/3264113/109515018-86356600-7a9e-11eb-9aac-c4bd8be5be48.png)


Note that this shouldn't have a close/cancel/cta as it should block the user app access.
note two: modal probably wants some info on _how_ to update.

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_
